features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Blacklist/List.jsx

```diff
@@ -95,6 +98,77 @@ export default function ListBlacklistThrottlingPolicies() {
         }),
     };
 
+    /**
+ * Mock API call
+ * @returns {Promise}.
+ */
+    function apiCall() {
+        let policyList;
+        let incrementId = 0;
+        return new Promise(((resolve, reject) => {
+            restApi.blacklistPoliciesGet().then((result) => {
+                policyList = result.body.list;
+                const blacklistPolicies = policyList.map((obj) => {
+                    let array = [];
+                    incrementId++;
+                    if (obj.conditionValue === Object(obj.conditionValue)) {
+                        Object.keys(obj.conditionValue);
+                        Object.values(obj.conditionValue);
+                        array = Object.entries(obj.conditionValue);
+                    } else {
+                        array.push(obj.conditionValue);
+                    }
+                    return {
+                        conditionId: incrementId,
+                        conditionUUID: obj.conditionId,
+                        conditionType: obj.conditionType,
+                        conditionValue: array,
+                        conditionStatus: obj.conditionStatus,
+                    };
+                });
+                setBlacklistPolicyList(blacklistPolicies);
+                resolve(blacklistPolicies);
+            }).catch((error) => {
+                reject(error);
+            });
+        }));
+    }
+
+    const handleConditionStatus = (conditionUUID, event) => {
+        let editedConditionId;
+        let editedConditionStatus;
+        const blacklistPolicyListNew = cloneDeep(blacklistPolicyList);
+        blacklistPolicyListNew.map((res) => {
```

Comment:
I think, If you want to change the elements in the `blacklistPolicyListNew ` using the event data, It's better to return modified element in the map and use the return value from the map and set it to state, by that way you might not need to `cloneDeep` it beforehand

Reply:
If i do it as mentioned above the all the Switch items flick when I change one, but if I clone deep this doesnt happen
![May-27-2020 18-35-35](https://user-images.githubusercontent.com/19324135/83022297-f19e6300-a048-11ea-95a2-a265d34dc042.gif)